lib/fs.js

```diff
@@ -2009,15 +2009,34 @@ function ReadStream(path, options) {
   this.closed = false;
 
   if (this.start !== undefined) {
-    if (typeof this.start !== 'number' || Number.isNaN(this.start)) {
+    // Case 1: If the type of start is not 'number', throw ERR_INVALID_ARG_TYPE
+    // (TypeError).
+    if (typeof this.start !== 'number') {
       throw new ERR_INVALID_ARG_TYPE('start', 'number', this.start);
     }
-    if (this.end === undefined) {
-      this.end = Infinity;
-    } else if (typeof this.end !== 'number' || Number.isNaN(this.end)) {
-      throw new ERR_INVALID_ARG_TYPE('end', 'number', this.end);
+
+    // Case 2: If start is NaN, throw ERR_OUT_OF_RANGE (RangeError).
+    if (Number.isNaN(this.start)) {
+      throw new ERR_OUT_OF_RANGE('start', 'not NaN', this.start);
+    }
+
+    // Case 3: If start is negative, throw ERR_OUT_OF_RANGE (RangeError).
+    if (this.start < 0) {
+      throw new ERR_OUT_OF_RANGE('start', 'positive', this.start);
     }
 
+    // Case 4: If start is fractional, round them to the nearest whole number.
+    // (Not integer as negatives not allowed)
```

Comment:
I’m a bit confused by the comment here – we do round to the nearest integer, right?